### PR TITLE
Initial commit

### LINKED ISSUE #301: Create Helm chart for GatewayD
https://github.com/gatewayd-io/k8s-experiments

### PR DESCRIPTION
**Pull request by ghost**
## Description

Initial commit that provides a basic helm chart for GatewayD.

### CODE REVIEW

File: charts/gatewayd/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+apiVersion: v2
+name: gatewayd
+description: Cloud-native database gateway and framework for building data-driven applications, Like API gateways, for databases.
+type: application
+version: 0.1.0
+appVersion: v0.8.3
+home: https://www.gatewayd.io
+icone: https://github.com/gatewayd-io/docs/blob/main/assets/gatewayd-logotype-light.png
```
Comment (mostafa) [inline charts/gatewayd/Chart.yaml]:
```suggestion
icon: https://github.com/gatewayd-io/docs/blob/main/assets/gatewayd-logotype-light.png
```

File: charts/gatewayd/templates/tests/test-connection.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  labels:
+    {{- include "gatewayd.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": test
+spec:
+  containers:
+    - name: wget
+      image: busybox
+      command: ['wget']
+      args: ['{{ include "gatewayd.fullname" . }}:{{ .Values.service.port }}']
```
Comment (mostafa) [inline charts/gatewayd/templates/tests/test-connection.yaml:14]: Is this where the healthz endpoints should be used? The healthz endpoints are released in [v0.8.4](https://github.com/gatewayd-io/gatewayd/releases/tag/v0.8.4).

**Review by mostafa — COMMENTED** [review]
Thanks for your contribution! :pray: 

I had a comment and a question.

File: charts/gatewayd/templates/tests/test-connection.yaml (hunk 2 above)
Comment (ghost) [reply charts/gatewayd/templates/tests/test-connection.yaml:14]: No it's not related to healthz endpoint (Liveness and Readiness). It's only used with `helm test` command. This chart is developed base on [v0.8.3](https://github.com/gatewayd-io/gatewayd/releases/tag/v0.8.3). In next release I'll add health endpoints to the chart.

**Review by mostafa — APPROVED** [review]
LGTM! 🎉

I'll create another ticket for improvement ideas.

**Pull request merged**